### PR TITLE
chore: add release script

### PR DESCRIPTION
**Pull request by balazsorban44**
- [x] Get a list of commits since the last tag
- [x] Early bailout ([skip ci] or chore(release): "bump version")
- [x] Filter out commits that did not touch package code
- [x] Group commits by package
- [x] For each package
   - [x] Group commits by type (Features, Bug fixes, Other)
   - [x] Determine the next version
   - [x] Update package.json with the new version
   - [x] Create the changelog
   - [x] Create git tag/release
- [x] Commit changes
- [x] Push git tag/release/commits
- [x] Push to npm
~- [ ] create `experimental` from PRs~ might to in a follow-up PR. For now, we can still publish at least `next-auth` with the old flow.
- [x] add dry run

partially solves #3952

### CODE REVIEW

**Comment by balazsorban44** [comment]
I think this is ready!

File: scripts/release/analyze.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+import path from "path"
+import { execSync } from "child_process"
+// @ts-ignore
+import jsonfile from "jsonfile"
+import semver from "semver"
+import parseCommit from "@commitlint/parse"
+// @ts-ignore
+import gitLog from "git-log-parser"
+// @ts-ignore
```
Comment (ThangHuuVu) [inline scripts/release/analyze.ts:12]: Trivial: you can add `@types/stream-to-array` and `@types/jsonfile` and remove `// @ts-ignore`

File: scripts/release/analyze.ts [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        else acc.other.push(commit)
+        return acc
+      },
+      { features: [], bugfixes: [], other: [] } as Record<
+        "features" | "bugfixes" | "other",
+        Commit[]
+      >
+    )
+
+    const featuresHasBreaking = grouppedPackage.features.some((c) =>
```
Comment (ThangHuuVu) [inline scripts/release/analyze.ts:147]: typo `grouped` 
```suggestion
    const featuresHasBreaking = groupedPackage.features.some((c) =>
```

**Review by ThangHuuVu — COMMENTED** [review]
Added some comments

**Review by ndom91 — APPROVED** [review]
LGTM :+1:

**Pull request merged**